pyasdf/block.py

```diff
@@ -521,6 +528,46 @@ def compression(self, compression):
     def is_compressed(self):
         return self._compression is not None
 
+    @property
+    def checksum(self):
+        return self._checksum
+
+    def _set_checksum(self, checksum):
+        if checksum == b'\0' * 16:
+            self._checksum = None
+        else:
+            self._checksum = checksum
+
+    def _calculate_checksum(self, data):
+        m = hashlib.new('md5')
+        m.update(self.data)
```

Comment:
Yeah -- at the moment, we just write the array in one go (which obviously does chunking somewhere along the line underneath), but that doesn't give us a good spot in *our* code to do the checksum calculation on the side.  I might defer that for now in that case...